File: C/1351-count-negative-numbers-in-a-sorted-matrix/main.go
package main

import "fmt"

func countNegatives(grid [][]int) int {
	r := len(grid) - 1
	c := 0

	count := 0
	for r >= 0 && c < len(grid[0]) {
		if grid[r][c] < 0 {
			count += len(grid[0]) - c
			r--
		} else {
			c++
		}
	}
	return count
}

func main() {
	fmt.Println("a")
}

File: C/1351-count-negative-numbers-in-a-sorted-matrix/main.rs
impl Solution {
  pub fn count_negatives(grid: Vec<Vec<i32>>) -> i32 {
    (0..grid.len())
      .fold((0, grid[0].len() as i32 - 1), |(ans, col), row| {
        let mut c = col as i32;
        while c >= 0 && grid[row][c as usize] < 0 {
          c -= 1;
        }
        (ans + grid[0].len() as i32 - 1 - c, c)
      })
      .0
  }
}
